src/main/java/ch/njol/skript/lang/function/Functions.java

```diff
@@ -146,18 +137,13 @@ public static Function<?> loadFunction(Script script, SectionNode node) {
 	/**
 	 * Loads the signature of function from given node.
 	 * @param script Script file name (<b>might</b> be used for some checks).
-	 * @param node Section node.
+	 * @param local If the signature of function is local
+	 * @param matchResult The match result of the function regex pattern
 	 * @return Signature of function, or null if something went wrong.
 	 */
 	@Nullable
-	public static Signature<?> loadSignature(String script, SectionNode node) {
-		SkriptLogger.setNode(node);
-		String key = node.getKey();
-		String definition = ScriptLoader.replaceOptions(key == null ? "" : key);
-		Matcher m = functionPattern.matcher(definition);
-		if (!m.matches())
-			return signError(INVALID_FUNCTION_DEFINITION);
-		String name = "" + m.group(1);
+	public static Signature<?> loadSignature(String script, boolean local, MatchResult matchResult) {
```

Comment:
I'm a bit unsure about this change - at least the MatchResult part. What about something like `String, boolean SectionNode` (script, local, node)?